docs(request-response): update docs about peer address management

wowinter13 opened the pull request:
## Description


Fixes https://github.com/libp2p/rust-libp2p/issues/5372

<!--
Please write a summary of your changes and why you made them.
This section will appear as the commit message after merging.
Please craft it accordingly.
For a quick primer on good commit messages, check out this blog post: https://cbea.ms/git-commit/

Please include any relevant issues here, for example:

Related https://github.com/libp2p/rust-libp2p/issues/ABCD.
Fixes https://github.com/libp2p/rust-libp2p/issues/XYZ.
-->

## Notes & open questions

<!--
Any notes, remarks, or open questions you have to make about the PR that don't need to go into the final commit message.
-->

Or we can maybe mention `Swarm::add_peer_address`

## Change checklist

<!-- Please add a Changelog entry in the appropriate crates and bump the crate versions if needed. See <https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

wowinter13 committed:
chore(docs): remove outdated note

dariusc93 reviewed (dismissed):
Thanks for the PR. While im not against removing it, it might make more sense to remove it upon `Behaviour::add_address` and `Behaviour::remove_address` being removed.

dariusc93 committed:
Merge branch 'master' into remove-outdated-note

dariusc93 committed:
chore: Add edition to workspace Cargo.toml and inherit workspace edition

Add edition to workspace Cargo.toml and Inherits workspace edition. Additionally, inherits rust-version in `libp2p-upnp` and `libp2p-websocket-websys`.

Pull-Request: #5896.

Khwahish29 committed:
fix: rename InsufficientPeers to NoPeersSubscribedToTopic for clarity

The `InsufficientPeers` error was unclear about whether it meant no peers were connected at all or no peers were subscribed to the topic. Rename it to `NoPeersSubscribedToTopic` to make it clear that this error occurs when trying to publish to a topic that has no subscribed peers.
Fixes #5907

Pull-Request: #5912.

wowinter13 committed:
chore(docs): fix outdated note

wowinter13 committed:
Merge branch 'master' into remove-outdated-note

elenaf9 commented on protocols/request-response/src/lib.rs:
```diff
@@ -430,8 +430,9 @@ where
     /// > address discovery, or known addresses of peers must be
-    /// > managed via [`Behaviour::add_address`] and
-    /// > [`Behaviour::remove_address`].
+    /// > managed via [`Swarm::add_peer_address`].
```
```suggestion
    /// > managed via [`libp2p_swarm::Swarm::add_peer_address`].
```
I think you have to provide the full path for rustdoc.

wowinter13 replied:
Thanks!
Sorry, sometimes I only have a bit of time per day for OSS and may forget to check the CI
@elenaf9

wowinter13 committed:
chore(docs): fix rustdoc refs

wowinter13 committed:
Merge branch 'master' into remove-outdated-note

elenaf9 committed:
Merge branch 'master' into remove-outdated-note

mergify[bot] committed:
Merge branch 'master' into remove-outdated-note
